chaincode/src/swaps/fetchTokenSwapByRequestId.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) Gala Games Inc. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { TokenSwapRequest } from "@gala-chain/api";
+
+import { GalaChainContext } from "../types";
+import { getObjectByKey } from "../utils";
+
+export async function fetchTokenSwapByRequestId(
+  ctx: GalaChainContext,
+  swapRequestId: string
+): Promise<TokenSwapRequest> {
+  let tokenSwap: TokenSwapRequest;
+
+  try {
+    tokenSwap = await getObjectByKey(ctx, TokenSwapRequest, swapRequestId);
+  } catch (error) {
+    throw new Error(`Token swap with swapRequestId ${swapRequestId} not found.`);
+  }
+
+  return tokenSwap;
```

Comment:
```suggestion
  return await getObjectByKey(ctx, TokenSwapRequest, swapRequestId);
```

why:
1. The error thrown indicates "not found" even if the underlying error in `getObjectByKey` is different
2. `Error` misses some useful properties that `ChainError` (thrown from `getObjectByKey`) has, like mapping to http 404.

I also don't think we need a custom error type for that function